### PR TITLE
OSSM-2006 Fix HasSynced() [maistra-2.2]

### PR DESCRIPTION
**Pull request by luksa**
Previously, if the set of namespaces in the multiNamespaceInformer was empty, HasSynced() would return true. This caused controllers to start, even though there are no informers in multiNamespaceeInformer. The controllers would read from the cache (which would return no resources), causing them to perform the reconcile on bad input data. Typically, this would cause the controller to remove resources that shouldn't be removed. Of course, moments later, the namespace list gets initialized, the controllers are notified of the appearance of new resources, and they recreate the resources they had removed a moment ago. Although the end state is correct, for some resource types, such as Routes, their deletion wreaks havoc.

HasSynced() shouldn't return true until SetNamespaces() has been called at least once (this is done by the ServiceMeshMemberRoll controller after it reads the ServiceMeshMemberRoll object for the first time). The informer also shouldn't be initialized with all namespaces, but should remain empty until the actual member namespace list is read from the ServiceMeshMemberRoll.

### CODE REVIEW

File: pkg/informers/namespace_set.go [hunk 1]
```diff
@@ -77,11 +78,20 @@ func (n *namespaceSet) List() []string {
...
+func (n *namespaceSet) Initialized() bool {
+	return n.initialized
+}
+
 // SetNamespaces replaces the set of namespaces.
 func (n *namespaceSet) SetNamespaces(namespaces ...string) {
 	n.lock.Lock()
 	defer n.lock.Unlock()
 
+	if !n.initialized {
```
Comment (jwendell) [inline pkg/informers/namespace_set.go:94]: Do we need atomicity here? e.g. using atomicBool instead?

Comment (luksa) [reply pkg/informers/namespace_set.go:94]: Good catch. I've updated `Initiliazed()` to also use `n.lock`.

**Review by jwendell — APPROVED** [review]

**Pull request merged**